lib/ci/tacos-apply

```diff
@@ -15,6 +15,32 @@ EOF
     exit 1
 fi
 
+if "$GITHUB_PR_DRAFT"; then
+    : no applying a draft! bail.
+    gha-step-summary >/dev/null <<EOF
+### TACOS Apply: $TF_ROOT_MODULE
+
+Applies are not allowed for draft PRs.
+Please set the PR ready for review first.
+
+Apply request refused: PR is a draft
+EOF
+    exit 1
+fi
+
+if [[ "$GITHUB_PR_STATE" != "OPEN" ]]; then
+    : no applying a closed PR! bail.
+    gha-step-summary >/dev/null <<EOF
+### TACOS Apply: $TF_ROOT_MODULE
+
+Applies are only allowed for open PRs.
+You may try reopening the PR first.
+
+Apply request refused: PR is not open
+EOF
+    exit 1
+fi
```

Comment:
We've hit the "rule of three" here -- time to DRY out with a function:

```
refuse() {
    reason="$1"
   (
   echo "### TACOS Apply: $TF_ROOT_MODULE"
   cat  # pass deeper explanation via stdin
   echo
   echo "Apply request refused: $reason"
EOF
) | gha-step-summary >/dev/null
    exit 1
}

...

refuse "PR is not open" <<EOF
Cannot apply closed PRs.
Please re-open the PR if it has not been applied.
EOF
```

Reply:
function added.